File: Lokesh_Saana/896-monotonic-array.py
# https://leetcode.com/problems/monotonic-array/

class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        inc=0
        dec=0
        for i in range(len(nums)-1):
            if nums[i]>nums[i+1]:
                dec=1
            if nums[i]<nums[i+1]:
                inc=1
            if inc==1 and dec==1:
                return False
        if inc==1 and dec==1:
                return False
        return True
        

File: Vindeep-Chaudhari/896_Monotonic_Array.cpp
/**
 * Problem Link - https://leetcode.com/problems/monotonic-array/
 */

class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        if(nums.size() == 1 | nums.size() == 2) return true;
        
        int nature = nums[1] - nums[0];
        for(int i = 1; i < nums.size()-1; i++) {
            int diff = nums[i+1] - nums[i];
            
            if(nature == 0) {
                nature = diff;
            } else {
                if(diff < 0 && nature > 0) return false;
                else if (diff > 0 && nature < 0) return false;
            }
        }
        
        return true;
    }
};